rust/build_assert.rs

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Build-time assertion.
+//!
+//! In Rust it is very common to use assertions extensively and defensively.
+//! Sometimes we know a condition could be checked statically, but it could not
+//! be enforced in Rust (e.g. perform some checks in const functions, but those
+//! functions could still be called in the runtime).
+//!
+//! This crate provides a method `build_error`, which will panic in
```

Comment:
Do we want to mention `build_error` (here and everywhere else)? Should a user ever call it?

Reply:
I think we do. It might be called after using checked math ops to unwrap Option.